### ISSUE TITLE
Called Variabled Is Not Reset When Variables To Use Lazy Query Hook Change

### ISSUE THREAD

**Issue by rossm6** [apollographql/apollo-client#8937]
**Intended outcome:**

When the variables passed to the useLazyQuery change the variable "called" returned by the hook should reset to false.

**Actual outcome:**

After the initial call of fetchInitial the "called" boolean flips to "true" and stays forever "true" as far as I can see.

**How to reproduce the issue:**

E.g.

```
    const variables = {
        first: PAGE_SIZE,
        after: '',
        orderBy: args?.orderBy,
        filters: {
            search: args?.filters?.search,
            status: args?.filters?.status || undefined
        }
    };

    /*
        
    */
    const [[fetchInitial, { data, loading, fetchMore, called }]] = useApolloHookAndHandleErrors(useLazyQuery, GET_YOUR_LISTINGS, {
        variables
    });

    useEffect(() => {
        let cmp = { orderBy: args?.orderBy, search: args?.filters?.search, status: args?.filters?.status }
        if (called && args && !deepEqual(previousRelevantArgs, cmp)) { // check called here
            fetchInitial({ variables });
        }
    }, [args,  called, previousRelevantArgs, fetchInitial, variables])

```

In the code snippet above I have to remove "called" from the IF statement.

In short - apollo js correctly lets us call fetchInitial multiple times if the variables change however it does not appear to reset "called" back to "false".

I haven't included a reproduction case because I haven't got the time right now and I doubt you guys will need it.

**Versions**

  System:
    OS: Linux 5.11 Ubuntu 20.04.3 LTS (Focal Fossa)
  Binaries:
    Node: 16.9.0 - ~/.nvm/versions/node/v16.9.0/bin/node
    Yarn: 1.22.15 - /usr/bin/yarn
    npm: 7.21.1 - ~/.nvm/versions/node/v16.9.0/bin/npm
  Browsers:
    Chrome: 94.0.4606.81
    Firefox: 93.0
  npmPackages:
    @apollo/client: ^3.4.15 => 3.4.16

**Comment by brainkim** [comment]
This is working as “designed.” The `called` property tends to be confusing. There’s some additional discussion here (https://github.com/apollographql/apollo-client/issues/7931), and in some other issues.

**Comment by rossm6** [comment]
My understanding is that "called" is a flag which tells us it "fetchInitial" has been called so we don't call fetchMore if not.  Is this not correct?

If it is then this is a bug I'd argue.

In short -

1. called is false
2. fetchInitial is called
3. called is true.  FetchInitial is now not callable.
4. variables passed to useLazyQuery change so fetchInitial is now callable again.
5. called should now be false.
